jax/_src/numpy/linalg.py

```diff
@@ -1220,13 +1220,17 @@ def norm(x: ArrayLike, ord: int | str | None = None,
                      " compute a vector-norm, or two axes to compute a matrix-norm.")
 
 @overload
-def qr(a: ArrayLike, mode: Literal["r"]) -> Array: ...
+def qr(a: ArrayLike,
+       mode: Literal["reduced", "complete", "raw", "full"] = "reduced",
+       ) -> QRResult: ...
 @overload
-def qr(a: ArrayLike, mode: str = "reduced") -> Array | QRResult: ...
+def qr(a: ArrayLike, mode: Literal["r"]) -> Array: ...
 
 @export
 @partial(jit, static_argnames=('mode',))
-def qr(a: ArrayLike, mode: str = "reduced") -> Array | QRResult:
+def qr(a: ArrayLike,
```

Comment:
Strictly speaking, the implementation should be untyped, but idk if mypy/pyright use the types from the overload definitions to type-check the implementation.